src/Conversion/ONNXToKrnl/ControlFlow/Loop.cpp

```diff
@@ -23,7 +23,9 @@ struct ONNXLoopOpLowering : public ConversionPattern {
 
   LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
       ConversionPatternRewriter &rewriter) const final {
-    auto loc = op->getLoc();
+    auto loc = NameLoc::get(
```

Comment:
Since this pattern is going to be used in a lot of place, would you be able to make it a templated function
``` C++
template<OP_TYPE>
Location ONNXLoc(Operation *op) { return NameLoc::get(OP_TYPE::getOperationName(), op->getContext()),
         op->getLoc()); }
```
(apologies if there are C++ error in the code above). You may need another version where the op type is OP_TYPE as sometime it might be a typed operation.